### PR TITLE
Implemented FCM system

### PR DESCRIPTION
**Pull request by KakaMintian**
Implemented FCM system. Biz will send message to client when he/she accepts the request. Client will send message to biz upon reserve/cancel reservations.

We need to have firebase function server alive in order to make FCM working.
Every time you want to change FCMService.java you need to re-install the app.

### CODE REVIEW

File: app/src/main/java/com/example/coen268/CreateAccountActivity.java [hunk 1]
```diff
@@ -39,6 +39,8 @@
     private FirestoreService firestoreService;
     private boolean mBound;
 
+    DatabaseReference m_databaseFcmRegToken;
```
Comment (m-cai) [inline app/src/main/java/com/example/coen268/CreateAccountActivity.java]: What is this reference for?

Comment (KakaMintian) [reply app/src/main/java/com/example/coen268/CreateAccountActivity.java]: Sorry, this should not be needed anymore. I'll fix it

File: app/src/main/java/com/example/coen268/LoginActivity.java [hunk 2]
```diff
@@ -207,13 +214,18 @@ public void onComplete(@NonNull Task<AuthResult> task) {
...
                                 Intent intent = new Intent(LoginActivity.this, CreateAccountActivity.class);
                                 intent.putExtra(Constants.KEY_ACCOUNT_TYPE, accountType);
                                 intent.putExtra(Constants.KEY_IS_GOOGLE_AUTH, true);
                                 startActivity(intent);
                             } else {
                                 FirebaseUser user = mAuth.getCurrentUser();
+
+                                // Send logged in uid to fcm service
+                                Intent intent = new Intent(FCMService.FCM_CMDCHANNEL);
+                                intent.putExtra("uid", user.getUid());
```
Comment (m-cai) [inline app/src/main/java/com/example/coen268/LoginActivity.java:227]: Does `intent.putExtra("Command", FCMService.CMD_LOGIN);` need to be here too? I noticed it was in the similar code above

Comment (KakaMintian) [reply app/src/main/java/com/example/coen268/LoginActivity.java:227]: It doesn't need to be here; I put it here just for convenience. Could be cached to a instance var, but it's not necessary.

**Pull request merged**